celements-scheduler/src/main/java/com/celements/scheduler/classdefs/SchedulerJobClass.java

```diff
@@ -0,0 +1,92 @@
+package com.celements.scheduler.classdefs;
+
+import org.springframework.stereotype.Component;
+import org.xwiki.model.reference.ClassReference;
+
+import com.celements.marshalling.EnumMarshaller;
+import com.celements.model.classes.AbstractClassDefinition;
+import com.celements.model.classes.fields.ClassField;
+import com.celements.model.classes.fields.CustomStringField;
+import com.celements.model.classes.fields.LargeStringField;
+import com.celements.model.classes.fields.StringField;
+import com.xpn.xwiki.XWikiConstant;
+
+@Component(SchedulerJobClass.CLASS_DEF_HINT)
+public class SchedulerJobClass extends AbstractClassDefinition
+    implements SchedulerJobClassDefinition {
+
+  public static final String DOC_NAME = "SchedulerJobClass";
+  public static final String XWIKI_SPACE = XWikiConstant.XWIKI_SPACE;
+  public static final String CLASS_DEF_HINT = XWIKI_SPACE + "." + DOC_NAME;
+  public static final ClassReference CLASS_REF = new ClassReference(XWIKI_SPACE, DOC_NAME);
+
+  public enum Status {
+    NORMAL, NONE
+  }
+
+  public static final ClassField<String> FIELD_JOB_NAME = new StringField.Builder(
+      CLASS_REF, "jobName")
+          .prettyName("Job Name")
+          .size(60)
+          .build();
+
+  public static final ClassField<String> FIELD_JOB_DESCRIPTION = new LargeStringField.Builder(
+      CLASS_REF, "jobDescription")
+          .rows(10)
+          .prettyName("Job Description")
+          .size(45)
+          .build();
+
+  public static final ClassField<String> FIELD_JOB_CLASS = new StringField.Builder(
+      CLASS_REF, "jobClass")
+          .prettyName("Job Class")
+          .size(60)
+          .build();
+
+  public static final ClassField<Status> FIELD_STATUS = new CustomStringField.Builder<>(
+      CLASS_REF, "status", new EnumMarshaller<>(Status.class, (e -> e.name().toLowerCase())))
```

Comment:
I like the idea of backing the status field with an Enum. Yet, this has to take in to account that the status field is mimicing the persistency of the `com.celements.scheduler.job.JobState` (see `com.celements.scheduler.SchedulerPlugin.scheduleJob(BaseObject, XWikiContext)` line 227 and f.f. esp. line 243, 245 and 251). Hence we need the following Status for completeness:
-   public static final String STATE_NORMAL = "Normal";
-   public static final String STATE_PAUSED = "Paused";
-   public static final String STATE_BLOCKED = "Blocked";
-   public static final String STATE_COMPLETE = "Complete";
-   public static final String STATE_ERROR = "Error";
-   public static final String STATE_NONE = "None";
AND they must be Pascal case (or too called Upper Camel) see e.g. line 241 in SchedulerPlugin class:
`          if (object.getStringValue("status").equals("Paused")) {`
Moreover use https://guava.dev/releases/23.0/api/docs/com/google/common/base/CaseFormat.html#UPPER_CAMEL to convert to Pascal case.